surface/type.go

```diff
@@ -26,11 +26,20 @@ func (s *Type) FieldWithName(name string) *Field {
 	if s == nil || s.Fields == nil || name == "" {
 		return nil
 	}
+	// Compares Go specific field names.
```

Comment:
With the change that you made below, I wonder if we should keep this test.

It would be nice if anything language-specific could be factored out of this layer.

Reply:
Yes, I was also thinking that. However, I checked gnostic-go-generator where this method is [used](https://github.com/googleapis/gnostic-go-generator/blob/ab235227442a8031d4a2dac629b992fa1c0605d7/render_server.go#L122) with language-specific values.

Reply:
OK, kudos for confirming that :) Let's go ahead like this and revisit the language-specific values later.